EventListener/IsGrantedListener.php

```diff
@@ -36,41 +41,92 @@ public function __construct(ArgumentNameConverter $argumentNameConverter, Author
         $this->authChecker = $authChecker;
     }
 
-    public function onKernelControllerArguments(FilterControllerArgumentsEvent $event)
+    public function onKernelController(FilterControllerEvent $event)
     {
         $request = $event->getRequest();
         /** @var $configurations IsGranted[] */
         if (!$configurations = $request->attributes->get('_is_granted')) {
             return;
         }
 
-        if (null === $this->authChecker) {
-            throw new \LogicException('To use the @IsGranted tag, you need to install symfony/security-bundle and configure your security system.');
+        $this->warnAboutAuthChecker();
+
+        foreach ($configurations as $configuration) {
+            // Only configurations WITHOUT subject.
+            if (null === $configuration->getSubject()) {
+                $this->checkIsGranted($configuration, null);
+            }
         }
+    }
+
+    public function onKernelControllerArguments(FilterControllerArgumentsEvent $event)
```

Comment:
this type hints a deprecated class, isn't it?
better remove the type hint and replace it with an annotation with the non-deprecated type
that'll preserve compat with 3.4 while preparing for 5.0
please check other event too